sdk/include/platform/ibex/platform-dma.hh

```diff
@@ -0,0 +1,118 @@
+#pragma once
+
+#define MALLOC_QUOTA 0x100000
+#define TEST_NAME "Allocator"
+#include "tests.hh"
+
+#include <cheri.hh>
+#include <compartment-macros.h>
+#include <cstdint>
+#include <stdint.h>
+#include <utils.hh>
+
+#include <thread.h>
+#include <thread_pool.h>
+
+using thread_pool::async;
+DECLARE_AND_DEFINE_ALLOCATOR_CAPABILITY(dmaDriverHeap, 8144);
+
+namespace Ibex {
+    class GenericDMA {
+        private:
+
+        Timeout noWait{0};
+        SObjStruct *dmaHeap;
+
+        struct StreamingInterface
+        {
+            uint32_t control;
+            uint32_t status;
+            uint32_t address;
+            uint32_t lengthInBytes;
```

Comment:
It would be good to have doc comments for these to explain what they are (but thanks for giving them human-friendly names!).